server/lib/scene/scene.actions.js

```diff
@@ -27,7 +51,14 @@ const actionsFunc = {
       device = self.stateManager.get('device', action.device);
       deviceFeature = getDeviceFeature(device, action.feature_category, action.feature_type);
     }
-    return self.device.setValue(device, deviceFeature, action.value);
+
+    const value = action.value || evaluate(Handlebars.compile(action.evaluate_value)(scope).replace(/\s/g, ''));
```

Comment:
Be careful with implicit conversion! 

In JS, 0 = false

If value = 0, this code will evaluate the second part (and probably crash)

I'll strongly suggest using strict checks like:

```javascript
let value;
if(action.evaluate_value !== undefined) {
  value = evaluate(Handlebars.compile(action.evaluate_value)(scope).replace(/\s/g, ''));
} else {
 value = action.value;
}
```

Reply:
Well seen
Note: It might be interesting to have a more recent version of ES to use new operators as nullish operator `??`

Reply:
I created a topic on the forum for that: https://community.gladysassistant.com/t/frontend-amelioration-diverses-experience-developpement/8125